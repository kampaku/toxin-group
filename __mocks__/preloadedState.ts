const currentRooms = [
  {
    accessibility: { wideCorridor: true, assistant: false },
    additionalFacilities: {
      breakfast: true,
      crib: false,
      feedingChair: false,
      shampoo: false,
      tv: false,
      writingDesk: false,
    },
    freeDays: {
      from: '2022-10-11T06:17:02.467Z',
      to: '2022-10-18T00:20:15.895Z',
    },
    guests: { kids: 2, baby: 0, adults: 4 },
    id: 10,
    imgSrc: [
      'https://firebasestorage.googleapis.com/v0/b/toxin-…=media&token=43bbc6bb-a9a8-4731-ba0e-89784af2195e',
      'https://firebasestorage.googleapis.com/v0/b/toxin-…=media&token=43bbc6bb-a9a8-4731-ba0e-89784af2195e',
      'https://firebasestorage.googleapis.com/v0/b/toxin-…=media&token=43bbc6bb-a9a8-4731-ba0e-89784af2195e',
      'https://firebasestorage.googleapis.com/v0/b/toxin-…=media&token=43bbc6bb-a9a8-4731-ba0e-89784af2195e',
    ],
    number: 111,
    price: 11423,
    rating: 5,
    reviewsAmount: 99,
    roomAmenities: { bedrooms: 2, beds: 4, bathrooms: 0 },
    rules: { allowSmoke: false, allowGuests: true, allowPets: true },
    type: 'Люкс',
  },
  {
    accessibility: { wideCorridor: true, assistant: false },
    additionalFacilities: {
      breakfast: true,
      crib: false,
      feedingChair: false,
      shampoo: false,
      tv: false,
      writingDesk: false,
    },
    freeDays: {
      from: '2022-10-11T06:17:02.467Z',
      to: '2022-10-18T00:20:15.895Z',
    },
    guests: { kids: 2, baby: 0, adults: 4 },
    id: 20,
    imgSrc: [
      'https://firebasestorage.googleapis.com/v0/b/toxin-…=media&token=43bbc6bb-a9a8-4731-ba0e-89784af2195e',
      'https://firebasestorage.googleapis.com/v0/b/toxin-…=media&token=43bbc6bb-a9a8-4731-ba0e-89784af2195e',
      'https://firebasestorage.googleapis.com/v0/b/toxin-…=media&token=43bbc6bb-a9a8-4731-ba0e-89784af2195e',
      'https://firebasestorage.googleapis.com/v0/b/toxin-…=media&token=43bbc6bb-a9a8-4731-ba0e-89784af2195e',
    ],
    number: 222,
    price: 11423,
    rating: 5,
    reviewsAmount: 99,
    roomAmenities: { bedrooms: 2, beds: 4, bathrooms: 0 },
    rules: { allowSmoke: true, allowGuests: true, allowPets: false },
    type: 'Люкс',
  },
];

const filters = {
  accessibility: {
    wideCorridor: false,
    assistant: false,
  },
  additionalFacilities: {
    breakfast: false,
    crib: false,
    feedingChair: false,
    shampoo: false,
    tv: false,
    writingDesk: false,
  },
  freeDays: {
    from: null,
    to: null,
  },
  guests: { kids: 0, adults: 0, baby: 0 },
  price: { from: 5000, to: 10000 },
  roomAmenities: { bedrooms: 0, beds: 0, bathrooms: 0 },
  rules: { allowGuests: false, allowSmoke: false, allowPets: false },
};

const user = {
  id: '',
  name: '',
  surname: '',
  email: '',
  birthday: '',
  gender: '',
  getSpecOffers: false,
  likes: [],
};
  
const booking = {
  data: [
    {
      date: { from: '21.08.2022', to: '31.08.2022' },
      guests: { adults: 2, baby: 2, kids: 2 },
      totalCost: 2000,
      imgSrc: [''],
      number: 123,
      price: 20000,
      rating: 3,
      reviewsAmount: 10,
      type: '',
    },
  ],
  status: 'fulfilled',
};

export const preloadedState = {
  rooms: { current: currentRooms },
  searchFilters: filters,
  user,
  booking
};
