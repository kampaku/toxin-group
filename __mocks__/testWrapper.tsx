import React, { PropsWithChildren } from 'react';
import { configureStore, PreloadedState } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { NextRouter } from 'next/router';
import { RouterContext } from 'next/dist/shared/lib/router-context';
import { render, RenderOptions } from '@testing-library/react';

import { createMockRouter } from './createMockRouter';

import { combinedReducer, RootState } from '../src/redux/store';

const setupStore = (preloadedState?: PreloadedState<RootState>) => {
  return configureStore({
    reducer: combinedReducer,
    preloadedState,
  });
};

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: ReturnType<typeof setupStore>;
}

const renderWithProviders = (
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {},
  router: Partial<NextRouter> = {}
) => {
  const Wrapper = ({ children }: PropsWithChildren<{}>): JSX.Element => {
    return (
      <Provider store={store}>
        <RouterContext.Provider value={createMockRouter(router)}>
          {children}
        </RouterContext.Provider>
      </Provider>
    );
  };
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

export { renderWithProviders };
