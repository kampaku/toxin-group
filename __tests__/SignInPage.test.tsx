import { fireEvent, screen } from '@testing-library/react';

import { renderWithProviders } from '../__mocks__/testWrapper';
import { preloadedState } from '../__mocks__/preloadedState';
import SignInPage from '../src/pages/sign-in/index';

const mockAuth = jest.fn().mockReturnThis();

const expectedProps = {
  profileName: '',
  makeAuth: mockAuth,
};

beforeEach(() => {
  jest.clearAllMocks();
});

const signInPage = renderWithProviders(
  <SignInPage {...expectedProps} />,
  { preloadedState },
  { asPath: '/signin' }
);

describe('signIn page snapshot', () => {
  it('draw page', () => {
    const { container } = signInPage;
    expect(container).toMatchSnapshot();
  });
});

describe('signIn page ', () => {
  it('render header', () => {
    const { container } = renderWithProviders(
      <SignInPage {...expectedProps} />,
      { preloadedState },
      { asPath: '/signin' }
    );
    const header = container.getElementsByTagName('header')[0];
    expect(header).toBeVisible();
  });

  it('render footer', () => {
    const { container } = renderWithProviders(
      <SignInPage {...expectedProps} />,
      { preloadedState },
      { asPath: '/signin' }
    );
    const header = container.getElementsByTagName('footer')[0];
    expect(header).toBeVisible();
  });

  it('render SignIn Form', () => {
    renderWithProviders(
      <SignInPage {...expectedProps} />,
      { preloadedState },
      { asPath: '/signin' }
    );
    const formName = screen.getByRole('heading', { name: /title/i });
    const signInBtn = screen.getByRole('button', { name: 'buttonSignIn' });

    expect(formName).toBeVisible();
    expect(signInBtn).toBeVisible();
  });

  it('call a function getAuth to sign in a user', async () => {
    const { container, getByText } = renderWithProviders(
      <SignInPage {...expectedProps} />,
      { preloadedState },
      { asPath: '/signin' }
    );

    fireEvent.change(container.querySelector(`[type="email"]`)!, {
      target: { value: 'someemail@mail.ru' },
    });
    fireEvent.change(container.querySelector(`[type="password"]`)!, {
      target: { value: 'SomePassword123' },
    });
    fireEvent.click(getByText('buttonSignIn'));

    await expect(mockAuth).toHaveBeenCalledTimes(1);
  });

  it('do not call a function getAuth without email', () => {
    const { container, getByText } = renderWithProviders(
      <SignInPage {...expectedProps} />,
      { preloadedState },
      { asPath: '/signin' }
    );

    fireEvent.change(container.querySelector(`[type="email"]`)!, {
      target: { value: '' },
    });
    fireEvent.change(container.querySelector(`[type="password"]`)!, {
      target: { value: 'SomePassword123' },
    });
    fireEvent.click(getByText('buttonSignIn'));

    expect(mockAuth).toHaveBeenCalledTimes(0);
  });

  it('do not call a function getAuth without password', () => {
    const { container, getByText } = renderWithProviders(
      <SignInPage {...expectedProps} />,
      { preloadedState },
      { asPath: '/signin' }
    );

    fireEvent.change(container.querySelector(`[type="email"]`)!, {
      target: { value: 'someemail@mail.ru' },
    });
    fireEvent.change(container.querySelector(`[type="password"]`)!, {
      target: { value: '' },
    });
    fireEvent.click(getByText('buttonSignIn'));

    expect(mockAuth).toHaveBeenCalledTimes(0);
  });
});
