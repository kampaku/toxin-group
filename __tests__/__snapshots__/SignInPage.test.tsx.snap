// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signIn page snapshot draw page 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="header__content"
    >
      <div
        class="header__logo"
      >
        <a
          class="logo"
          href="/"
        >
          Next image stub
          Next image stub
        </a>
      </div>
      <div
        class="header__menu"
      >
        <div
          class="header__burgerMenu"
        >
          <nav
            class="burger"
          >
            <svg
              class="burger__icon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
              />
            </svg>
          </nav>
        </div>
        <div
          class="header__userMenu"
        >
          <div
            class="authentication"
          >
            <div
              class="authentication__signIn"
            >
              <a
                class="button button_type_signIn"
                href="/sign-in"
              >
                <span
                  class="button_type_signInText"
                >
                  signIn
                </span>
              </a>
            </div>
            <div
              class="authentication__registration"
            >
              <a
                class="button button_type_registration"
                href="/register"
              >
                <span
                  class="button_type_signInText"
                >
                  signUp
                </span>
              </a>
            </div>
          </div>
        </div>
        <div
          class="header__navbar"
        >
          <ul
            class="navBar"
          >
            <li
              class="navLink"
            >
              <a
                class="navLink__button"
                href="/about"
                title="aboutUs"
              >
                aboutUs
              </a>
            </li>
            <li
              class="navLink navLink_type_dropdown"
            >
              <div
                class="navLink__dropdown"
              >
                <svg
                  class="navLink__icon"
                  fill="currentColor"
                  height="21px"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="21px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
                <div
                  class="navLink__button"
                  title="services"
                >
                  services
                </div>
              </div>
            </li>
            <li
              class="navLink"
            >
              <a
                class="navLink__button"
                href="/vacancies"
                title="vacancies"
              >
                vacancies
              </a>
            </li>
            <li
              class="navLink"
            >
              <a
                class="navLink__button"
                href="/news"
                title="news"
              >
                news
              </a>
            </li>
            <li
              class="navLink navLink_type_dropdown"
            >
              <div
                class="navLink__dropdown"
              >
                <svg
                  class="navLink__icon"
                  fill="currentColor"
                  height="21px"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="21px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
                <div
                  class="navLink__button"
                  title="agreements"
                >
                  agreements
                </div>
              </div>
            </li>
            <li
              class="languageSwitcher"
            >
              <div
                class="switcher"
              >
                <div>
                  <button
                    class="btn"
                    type="button"
                  >
                    Next image stub
                    <svg
                      class="arrow"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </header>
  <main>
    <div
      class="signInPage"
    >
      <div
        class="signInPage__container"
      >
        <form
          class="signInPageForm"
        >
          <div
            class="signInPageForm__inner"
          >
            <h2
              class="signInPageForm__title"
            >
              title
            </h2>
            <div
              class="signInPageForm__input"
            >
              <div
                class=""
              >
                <input
                  class="input"
                  name=""
                  placeholder="Email"
                  type="email"
                  value=""
                />
              </div>
            </div>
            <div
              class="signInPageForm__input"
            >
              <div
                class=""
              >
                <input
                  class="input"
                  name=""
                  placeholder="password"
                  type="password"
                  value=""
                />
              </div>
            </div>
            <div
              class="signInPageForm__button"
            >
              <button
                class="button button_type_submit"
                type="submit"
              >
                <span
                  class="button__icon"
                >
                  <svg
                    fill="currentColor"
                    height="24px"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="24px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                    />
                  </svg>
                </span>
                <span
                  class="button_type_signInText"
                >
                  buttonSignIn
                </span>
              </button>
            </div>
            <div
              class="signInPageForm__direction"
            >
              <p
                class="signInPageForm__text"
              >
                account
              </p>
              <div
                class="signInPageForm__link"
              >
                <a
                  class="button button_type_signIn button_in_authorization"
                  href="/Registration"
                >
                  <span
                    class="button_type_signInText"
                  >
                    buttonSignUp
                  </span>
                </a>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </main>
  <footer
    class="footer"
  >
    <div
      class="footerContent"
    >
      <div
        class="footerContent__container"
      >
        <div
          class="footerContent__content"
        >
          <div
            class="footerContent__column"
          >
            <div
              class="footerContent__logo"
            >
              <a
                class="logo"
                href="/"
              >
                Next image stub
                Next image stub
              </a>
            </div>
            <p
              class="footerContent__description"
            >
              description
            </p>
          </div>
          <div
            class="footerContent__column"
          >
            <div
              class="footerList__list"
            >
              <h3
                class="footerList__subtitle"
              >
                navigation
              </h3>
              <a
                class="footerList__link"
                href="/mock-address"
              >
                aboutUs
              </a>
              <a
                class="footerList__link"
                href="/mock-address"
              >
                news
              </a>
              <a
                class="footerList__link"
                href="/mock-address"
              >
                support
              </a>
              <a
                class="footerList__link"
                href="/mock-address"
              >
                services
              </a>
            </div>
          </div>
          <div
            class="footerContent__column"
          >
            <div
              class="footerList__list"
            >
              <h3
                class="footerList__subtitle"
              >
                aboutUs
              </h3>
              <a
                class="footerList__link"
                href="/mock-address"
              >
                aboutService
              </a>
              <a
                class="footerList__link"
                href="/mock-address"
              >
                ourTeam
              </a>
              <a
                class="footerList__link"
                href="/mock-address"
              >
                vacancy
              </a>
              <a
                class="footerList__link"
                href="/mock-address"
              >
                investors
              </a>
            </div>
          </div>
          <div
            class="footerContent__column"
          >
            <div
              class="footerList__list"
            >
              <h3
                class="footerList__subtitle"
              >
                support
              </h3>
              <a
                class="footerList__link"
                href="/mock-address"
              >
                agreements
              </a>
              <a
                class="footerList__link"
                href="/mock-address"
              >
                communities
              </a>
              <a
                class="footerList__link"
                href="/mock-address"
              >
                contactUs
              </a>
            </div>
          </div>
          <div
            class="footerContent__column"
          >
            <h3
              class="footerContent__subtitle"
            >
              subscription
            </h3>
            <p
              class="footerContent__description"
            >
              subscriptionText
            </p>
            <form
              class="footerContent__form"
            >
              <div
                class="input__wrapper"
              >
                <input
                  class="input input_withArrow"
                  name=""
                  placeholder="Email"
                  type="text"
                  value=""
                />
                <button
                  aria-label="send form"
                  class="input__button"
                  type="submit"
                >
                  <span
                    class="input__buttonInner"
                  />
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
    <div
      class="footerSocial"
    >
      <div
        class="footerSocial__container"
      >
        <div
          class="footerSocial__content"
        >
          <div
            class="footerSocial__logo"
          >
            <a
              class="logo"
              href="/"
            >
              Next image stub
              Next image stub
            </a>
          </div>
          <p
            class="footerSocial__text"
          >
            copyright
          </p>
          <div
            class="footerSocial__icons"
          >
            <a
              class="footerSocial__link"
              href="https://twitter.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Next image stub
            </a>
            <a
              class="footerSocial__link"
              href="https://facebook.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Next image stub
            </a>
            <a
              class="footerSocial__link"
              href="https://instagram.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Next image stub
            </a>
          </div>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
